x-pack/plugins/observability_solution/infra/public/components/asset_details/template/page.tsx

```diff
@@ -86,32 +83,15 @@ export const Page = ({ tabs = [], links = [] }: ContentTemplateProps) => {
       onboardingFlow={asset.type === 'host' ? OnboardingFlow.Hosts : OnboardingFlow.Infra}
       dataAvailabilityModules={DATA_AVAILABILITY_PER_TYPE[asset.type] || undefined}
       pageHeader={{
-        pageTitle: asset.name,
+        pageTitle: loading ? <EuiLoadingSpinner size="m" /> : asset.name,
```

Comment:
The "loading data" spinner was blocking the rendering of the rest of the page. the loading in this case will only be true in case the page still needs to fetch the asset name.